pkg/evaluation/manager.go

```diff
@@ -209,11 +213,132 @@ func (m *manager) EvaluateResource(ctx context.Context, request *pb.ResourceEval
+	}
+
+	var resourceEvaluation pb.ResourceEvaluation
+	err = protojson.UnmarshalOptions{DiscardUnknown: true}.Unmarshal(resourceEvaluationResponse.Hits.Hits[0].Source, &resourceEvaluation)
+	if err != nil {
+		return nil, util.GrpcInternalError(log, "error unmarshalling resource evaluation into json", err)
+	}
+
+	policyEvaluationResponse := searchResponse.Responses[1]
+
+	var policyEvaluations []*pb.PolicyEvaluation
+	for _, hit := range policyEvaluationResponse.Hits.Hits {
+		var policyEvaluation pb.PolicyEvaluation
+		err = protojson.UnmarshalOptions{DiscardUnknown: true}.Unmarshal(hit.Source, &policyEvaluation)
+		if err != nil {
+			return nil, util.GrpcInternalError(log, "error unmarshalling policy evaluation into json", err)
+		}
+
+		policyEvaluations = append(policyEvaluations, &policyEvaluation)
+	}
+
+	return &pb.ResourceEvaluationResult{
+		ResourceEvaluation: &resourceEvaluation,
+		PolicyEvaluations:  policyEvaluations,
+	}, nil
 }
 
 func (m *manager) ListResourceEvaluations(ctx context.Context, request *pb.ListResourceEvaluationsRequest) (*pb.ListResourceEvaluationsResponse, error) {
-	panic("implement me")
+	log := m.logger.Named("ListResourceEvaluations").With(zap.String("resourceUri", request.ResourceUri))
+
+	if request.ResourceUri == "" {
+		return nil, util.GrpcErrorWithCode(log, "resourceUri is required", nil, codes.InvalidArgument)
+	}
+
+	queries := filtering.Must{
+		&filtering.Query{
+			Term: &filtering.Term{
+				"resourceVersion.version": request.ResourceUri,
+			},
+		},
+	}
+
+	if request.Filter != "" {
+		filterQuery, err := m.filterer.ParseExpression(request.Filter)
+		if err != nil {
+			return nil, util.GrpcInternalError(log, "error parsing filter expression", err)
+		}
+
+		queries = append(queries, filterQuery)
+	}
+
+	searchRequest := &esutil.SearchRequest{
+		Index: m.indexManager.AliasName(constants.EvaluationsDocumentKind, ""),
+		Search: &esutil.EsSearch{
+			Query: &filtering.Query{
+				Bool: &filtering.Bool{
+					Must: &queries,
+				},
+			},
+			Sort: map[string]esutil.EsSortOrder{
+				"created": esutil.EsSortOrderDescending,
+			},
+		},
+	}
+
+	if request.PageSize != 0 {
+		searchRequest.Pagination = &esutil.SearchPaginationOptions{
+			Size:  int(request.PageSize),
+			Token: request.PageToken,
+		}
+	}
+
+	searchResponse, err := m.esClient.Search(ctx, searchRequest)
+	if err != nil {
+		return nil, util.GrpcInternalError(log, "error searching for resource evaluations", err)
+	}
+
+	var resourceEvaluations []*pb.ResourceEvaluation
+	for _, hit := range searchResponse.Hits.Hits {
+		var resourceEvaluation pb.ResourceEvaluation
+		err = protojson.UnmarshalOptions{DiscardUnknown: true}.Unmarshal(hit.Source, &resourceEvaluation)
+		if err != nil {
+			return nil, util.GrpcInternalError(log, "error unmarshalling resource evaluation into json", err)
+		}
+
+		resourceEvaluations = append(resourceEvaluations, &resourceEvaluation)
+	}
+
+	return &pb.ListResourceEvaluationsResponse{
```

Comment:
`ListResourceEvaluations` doesn't return the policy evaluations, but `GetResourceEvaluation` does. Should their behavior be the same? I'd imagine callers would be turning around to perform a `GetResourceEvaluation` after listing anyway, in order to get the policy evaluation results for at least on of the resource evaluations.

Reply:
I had thought about this too - I ended up going down this route because I figured this RPC was mostly for the UI, and we wouldn't need the policy evaluation details on the list view.  I can update this RPC to return a slice of `*pb.ResourceEvaluationResult` instead though.

Reply:
Returning  a slice of `*pb.ResourceEvaluationResult` sounds good to me. We can add the policy evaluations in if there's a need for it later.

Reply:
The `*pb.ResourceEvaluationResult` slice _would_ contain policy evaluation data in it though.  So I can keep the implementation as-is (no policy evaluation data), or update it to return a different data structure that includes policy evaluation data.  I'm fine with either.